src/_imaging.c

```diff
@@ -1175,59 +1178,62 @@ _getpixel(ImagingObject* self, PyObject* args)
     return getpixel(self->image, self->access, x, y);
 }
 
+#define HISTOGRAM_METHOD_PROLOGUE(HISTO)                                            \
```

Comment:
An internal function will look better than macros.

Reply:
I had used a macro to be as minimally invasive toward the existent code (without resorting to copy-pasting).

I am happy to refactor it as e.g. a `static` function resident in the `_imaging.c` TU, tho – can you do me a favor, and point me towards an exemplary internal-use function?

Reply:
@homm sweet! I can rewrite this and push a new change - I will most likely make a new pull request for it. Thank you!

Reply:
@homm See PR #3608, which incorporates this code. Thanks again!